aten/src/ATen/native/sparse/SparseCsrTensor.cpp

```diff
@@ -25,7 +25,6 @@ SparseCsrTensor new_csr_tensor(const TensorOptions& options) {
   if (options.device().is_cuda()) {
     dispatch_key = DispatchKey::SparseCsrCUDA;
   } else {
-    TORCH_INTERNAL_ASSERT(options.device().is_cpu());
```

Comment:
This assert still seems valid...

Reply:
I removed this assert because is too weak and added this instead https://github.com/pytorch/pytorch/blob/dd45bcfba054cedefbcc558c2b6a4859d322c3ff/aten/src/ATen/native/sparse/SparseCsrTensor.cpp#L49   

It covers cases when options.device() is kMeta for example.